cpp/src/arrow/python/python_to_arrow.cc

```diff
@@ -708,6 +709,26 @@ Status SerializeObject(PyObject* context, PyObject* sequence, SerializedPyObject
   return Status::OK();
 }
 
+Status SerializeTensor(std::shared_ptr<Tensor> tensor, SerializedPyObject* out) {
+  std::shared_ptr<Array> array;
+  SequenceBuilder builder(nullptr);
+  RETURN_NOT_OK(builder.AppendTensor(static_cast<int32_t>(out->tensors.size())));
+  out->tensors.push_back(tensor);
+  RETURN_NOT_OK(builder.Finish(nullptr, nullptr, nullptr, nullptr, &array));
+  out->batch = MakeBatch(array);
+  return Status::OK();
+}
+
+Status WriteTensorHeader(std::shared_ptr<DataType> dtype,
+                         const std::vector<int64_t>& shape, int64_t tensor_num_bytes,
+                         io::OutputStream* dst) {
+  auto empty_tensor = std::make_shared<arrow::Tensor>(
+      dtype, std::make_shared<arrow::Buffer>(nullptr, tensor_num_bytes), shape);
+  arrow::py::SerializedPyObject serialized_tensor;
```

Comment:
Are these explicit namespaces necessary? Tensor, Buffer, and SerializedPyObject should all be visible here

Reply:
Yeah, I'll change that (it was originally developed in the tensorflow op file and there were some name clashes and I forgot to remove the namespaces).